### ISSUE TITLE
【论文复现】torch.scatter_() 用法问题

### ISSUE THREAD

**Issue by Lieberk** [PaddlePaddle/Paddle#38813]
pytorch使用scatter_()实现了one-hot编码，举例如下：
![微信图片_20220108214504](https://user-images.githubusercontent.com/57614716/148647107-91185f35-5211-4867-a4e5-ebc0537498b8.png)
![微信图片_20220108214520](https://user-images.githubusercontent.com/57614716/148647112-14996a49-4e6c-43c1-9413-035d96db18f3.png)

目前paddle的scatter()和pytorch的scatter_()功能好像不一样，那如何使用paddle实现上述的one-hot编码。如果paddle没有对应的api，那怎样组合其他api可以实现上述的功能，麻烦老师解答一下，时间挺急的！

**Comment by zhaocaibei123** [comment]
paddle的scatter用法确实是不一样，请问是想用cross entropy loss吗，是的话paddle的cross_entropy label不需要扩充成one-hot编码，参考示例： https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/nn/functional/cross_entropy_cn.html#cross-entropy
另外，第二张图里的label是第一张图的label嘛，看起来似乎对应不上？

**Comment by Lieberk** [comment]
老师，就只实现上述的one-hot编码功能，在pytorch的scatter_()用法里第二张图的label是第一张图的label，有没有组合方式或者api实现上述的功能啊，困扰好久了。

**Comment by Lieberk** [comment]
> paddle的scatter用法确实是不一样，请问是想用cross entropy loss吗，是的话paddle的cross_entropy label不需要扩充成one-hot编码，参考示例： https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/nn/functional/cross_entropy_cn.html#cross-entropy 另外，第二张图里的label是第一张图的label嘛，看起来似乎对应不上？

老师，就只实现上述的one-hot编码功能，在pytorch的scatter_()用法里第二张图的label是第一张图的label，有没有组合方式或者api实现上述的功能啊，困扰好久了。

**Comment by zhaocaibei123** [comment]
paddle已在12月底新增了paddle.put_along_axis_ API对齐torch的scatter_算子，可以拉下develop分支，编译最新的代码来使用
put_along_axis_文档：https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/put_along_axis__cn.html#put-along-axis
